### PR TITLE
Scope new Establish Claim Task creation to decision type

### PR DESCRIPTION
**Pull request by shanear**
Every Establish Claim Task we tried to create today had multiple
appeals for the file number, which is annoying. The way to fix this
is to scope it to the type of decision.

Also, added refactored a little bit and added some better error
messaging.

### CODE REVIEW

File: app/services/appeal_repository.rb [hunk 1]
```diff
@@ -33,11 +33,21 @@ def self.load_vacols_data(appeal)
   end
 
   # :nocov:
-  def self.load_vacols_data_by_vbms_id(appeal)
+  def self.load_vacols_data_by_vbms_id(appeal:, decision_type:)
```
Comment (joofsh) [inline app/services/appeal_repository.rb]: This was formally a generic, reusable method (although with annoying limitations when multiple appeals). This version is definitely better but now only usable by appeals with a decision already. Can we either rename it to be specific to the type of appeal it searches, or maybe support a nil decision_type? I think having the `else` condition search all appeals like before may make sense.

Also, you're no longer joining to folder & correspondent (but we're setting true on vacols values fetched). I imagine the next dev that tries to use this method will get stumped on why it doesn't work properly. If you don't want to do the joins, maybe add a comment to explain?

File: app/services/appeal_repository.rb [hunk 2]
```diff
@@ -33,11 +33,21 @@ def self.load_vacols_data(appeal)
...
+                 else
+                   fail UnrecognizedDecisionTypeError
+                 end
+
     case_records = MetricsService.timer "loaded VACOLS case #{appeal.vbms_id}" do
-      VACOLS::Case.includes(:folder, :correspondent).where(bfcorlid: appeal.vbms_id).all
+      case_scope.where(bfcorlid: appeal.vbms_id)
     end
 
+    fail ActiveRecord::RecordNotFound if case_records.empty?
```
Comment (joofsh) [inline app/services/appeal_repository.rb]: nice 👍

File: app/services/create_establish_claim.rb [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+      establish_claim.prepare!
+    else
+      @error_message = "This appeal did not have a decision document in VBMS."
+    end
+
+  rescue MultipleAppealsByVBMSIDError
+    @error_message = "There were multiple appeals matching this VBMS ID."
+  rescue ActiveRecord::RecordNotFound
+    @error_message = "Appeal not found for that decision type." \
+      "Make sure to add the 'S' or 'C' to the end of the file number."
```
Comment (joofsh) [inline app/services/create_establish_claim.rb]: 👍

File: app/services/create_establish_claim.rb [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+  rescue UnrecognizedDecisionTypeError
+    @error_message = "You must select a decision type"
+  end
+
+  def create_appeal
```
Comment (joofsh) [inline app/services/create_establish_claim.rb]: This already exists in the [Appeal model](https://github.com/department-of-veterans-affairs/caseflow/blob/master/app/models/appeal.rb#L202-L208). Can we use that instead or delete it?

File: app/services/create_establish_claim.rb [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+  private
+
+  def perform_create_establish_claim
+    establish_claim = EstablishClaim.find_or_create_by(appeal: create_appeal)
+
+    # Admin has to confirm appeal has a decision document
+    if establish_claim.may_prepare?
+      establish_claim.prepare!
+    else
+      @error_message = "This appeal did not have a decision document in VBMS."
```
Comment (joofsh) [inline app/services/create_establish_claim.rb]: I don't think this is the right error message. The only reason you wouldn't be able to prepare the task is if it is not in the `unprepare` state (if the task already existed and was previously completed). Maybe an error message that tells the user the task's current state?

File: app/services/appeal_repository.rb (hunk 1 above)
Comment (shanear) [inline app/services/appeal_repository.rb]: Sounds good, I'll add the nil option.

Also `VACOLS::Case.amc_full_grants` and `VACOLS::Case.remands_ready_for_claims_establishment` have the joins included in them.

File: app/services/create_establish_claim.rb (hunk 4 above)
Comment (shanear) [inline app/services/create_establish_claim.rb]: It was deleted. See `app/models/appeal.rb` above on the PR.

File: app/services/create_establish_claim.rb (hunk 5 above)
Comment (shanear) [inline app/services/create_establish_claim.rb]: ohh interesting, i see now. Will change.

**Pull request merged**